### PR TITLE
Update to build and tests - fix seeking behaviour

### PR DESCRIPTION
**Pull request by nbirkenshaw-mux**
This updates our build system and tests for the roku devices, and fixes some of the seeking related behaviour at the same time.

The biggest change is removing usage of gulp - for whatever reason it disagreed violently with my machine, and our combination of dependencies was so out of date it was a question for optimizing to reduce critical security vulnerabilities in the packages with none not ever being an option. As such it was easier to rewrite that in go, and radically reduce our dependencies. It does mean users need golang installed so they can build the "test_runner" tool here with a "go build" in the current directory. Given this isn't wildly different to advanced usage of esbuild I think it's acceptable.

BSkyB's bslint got replaced with the rokucommunity one, which is actively maintained.

The actual task structure (now in the golang) is basically identical to the gulp, if not simplified and easier to reason about. For example, to test you would run:
`./test_runner test`
The only oddity of this is there is a pause between the deploy_test task and actually running the tests - this is to give the roku a chance to reach a stable state before telnetting into it to retrieve the results.

One set of tests, relating to stream identification, was completely out of date to the point of discarding. Other tests were fixed by actually fixing the underlying bugs.

VideoStateHandling got extended with new tests relating to the seeking behaviour problems. Generally I tried to move more towards what has worked with the Exoplayer tests. That resulted in more substantial changes to the event handling. I'm not convinced this is 100% right, but we can at least more easily tell what it's doing.

### CODE REVIEW

**Review by skidder — APPROVED** [review]

**Review by jsanford8 — COMMENTED** [review]
no straight approval yet, though this is _wonderful_. I have two comments, one on something I don't think matters really but can clean up (the main in package.json) but then another on versioning.

File: package.json [hunk 1]
```diff
@@ -1,42 +1,28 @@
 {
   "name": "roku-mux",
-  "version": "1.1.0",
-  "description": "Mux SDK for Roku",
-  "main": "none",
+  "version": "1.0.0",
```
[the remark this answers is not in the document]
Comment (nbirkenshaw-mux) [reply package.json]: Redid the package.json so it's only about pulling in the bslint file, as is now all that is needed. This means the version is only mentioned in one place, and no mention of gulp.

File: package.json [hunk 2]
```diff
@@ -1,42 +1,28 @@
...
+  "description": "mux analytics. SDK and Sample Application for testing.",
+  "main": "gulpfile.js",
```
[the remark this answers is not in the document]
Comment (nbirkenshaw-mux) [reply package.json]: Correct - see other comment re package.json

**Review by jsanford8 — APPROVED** [review]

**Comment by nbirkenshaw-mux** [comment]
Slightly odd one - I was at the point of releasing this PR but the SDK version in mux-analytics.brs is 1.2.1 and our git tagged release is 1.0.3 . . . 

I'd suggest moving everything to 1.2.1 and explaining this jump in the release notes and then trying to keep them in sync. Have I misinterpreted what the "MUX_SDK_VERSION" is for?

**Comment by skidder** [comment]
> I'd suggest moving everything to 1.2.1 and explaining this jump in the release notes and then trying to keep them in sync. Have I misinterpreted what the "MUX_SDK_VERSION" is for?

@nbirkenshaw-mux Absolutely, sounds like we missed a few git tags somewhere :)

**Pull request merged**